.gitlab-ci.yml

```diff
@@ -9,63 +9,89 @@
   SHARED_REPOS_DIR: "${USER_CI_TOP_DIR}/repos"
   AUTOTEST_ROOT: "${SHARED_REPOS_DIR}"
-  # MFEM_DATA_DIR is setup in '.gitlab/configs/setup-build-and-test.yml' and
-  # used in '.gitlab/configs/<machine>-config.yml':
+# MFEM_DATA_DIR is setup in '.gitlab/configs/setup-build-and-test.yml' and
+# used in '.gitlab/configs/<machine>-config.yml':
   MFEM_DATA_DIR: "${SHARED_REPOS_DIR}/mfem-data"
-
 # Defines the default choice for updating the saved baseline results. By default
 # the baseline can only be updated from the master branch. This variable offers
 # the option to manually ask for rebaselining from another branch if necessary.
-  REBASELINE: "NO"
-  AUTOTEST: "NO"
-  # AUTOTEST_COMMIT: used only when AUTOTEST is set to YES.
-  # * If AUTOTEST_COMMIT is NOT set to NO, reporting jobs will commit their
-  #   files to the MFEM/autotest repo.
-  # * If AUTOTEST_COMMIT is set to NO, reporting jobs will NOT commit their
-  #   files to the MFEM/autotest repo. Instead they will just show the contents
-  #   of the report files and remove them.
-  AUTOTEST_COMMIT: "YES"
+  REBASELINE: "OFF"
+  AUTOTEST: "OFF"
+# AUTOTEST_COMMIT: used only when AUTOTEST is set to ON.
+# * If AUTOTEST_COMMIT is NOT set to OFF, reporting jobs will commit their
+#   files to the MFEM/autotest repo.
+# * If AUTOTEST_COMMIT is set to OFF, reporting jobs will NOT commit their
+#   files to the MFEM/autotest repo. Instead they will just show the contents
+#   of the report files and remove them.
+  AUTOTEST_COMMIT: "ON"
 
-# Trigger subpipelines:
-ruby-build-and-test:
-  stage: sub-pipelines
-  variables:
-    # Explicitly pass down values that we want to be able to set when triggering
-    # pipelines manually or using scheduling
-    AUTOTEST: "${AUTOTEST}"
-    AUTOTEST_COMMIT: "${AUTOTEST_COMMIT}"
+##### SHARED_CI CONFIGURATION
+# Required information about GitHub repository
+  GITHUB_PROJECT_NAME: "mfem"
+  GITHUB_PROJECT_ORG: "MFEM"
+# Override the pattern describing branches that will skip the "draft PR filter
+# test".  Add protected branches here. See default value in
+# preliminary-ignore-draft-pr.yml.
+#  ALWAYS_RUN_PATTERN: ""
+
+###############################################################################
+##### High level stages
+# We organize the build-and-test stage with sub-pipelines. Each sub-pipeline
+# corresponds to a test batch on a given machine.
+stages:
+  - prerequisites
+  - build-and-test
+  - baseline #MFEM specific
+
+###############################################################################
+# Template for jobs triggering a build-and-test sub-pipeline:
+.build-and-test:
+  stage: build-and-test
   trigger:
-    include: .gitlab/ruby-build-and-test.yml
+    include:
+      - local: '.gitlab/custom-jobs-and-variables.yml'
+      - project: 'radiuss/radiuss-shared-ci'
+        ref: 'woptim/improve-job-cmd-handling'
+        file: 'pipelines/${CI_MACHINE}.yml'
+      - artifact: '${CI_MACHINE}-jobs.yml'
+        job: 'generate-job-lists'
```

Comment:
This is the heard of how the shared CI works. This is a template that will be used for each machine we want a pipeline for:
The sub-pipelines are generated by appending three files:
- the local `.gitlab/custom-jobs-and-variables.yml` is where you are going to configure the Shared CI for your needs.
- the remote `pipelines/${CI_MACHINE}.yml` is provided by RADIUSS Shared CI with up-to-date syntax and extra features.
- the `${CI_MACHINE}-jobs.yml`, generated by the `generate-job-lists` (see `.gitlab/subscribed-pipelines.yml`), lists all the MFEM defined jobs.

Note that RADIUSS provides a list of shared jobs that MFEM could include to build with toolchains identified as key by end-users.